chevah/python-package
# 2024.08 updates.

Scope
=====

Patch Python and OpenSSL for as many security issues as feasibly possible. Fixes #176 

Update libs and modules, if possible.

Changes
=======

Python security hot patches applied on all platforms for: CVE-2017-18207, CVE-2021-4189, CVE-2022-45061, CVE-2022-48565, CVE-2024-7592.

Patched Python 2.7.18 sources on non-Windows platforms for: CVE-2022-48560, CVE-2022-48566, CVE-2023-40217, CVE-2024-0397.

Patched OpenSSL 1.1.1w sources for: CVE-2023-5678, CVE-2024-0727,  CVE-2024-2511, CVE-2024-4741, CVE-2024-5535. 

Patched our `cryptography` sources for CVE-2023-49083.

Lib updates:
  * `libffi` to 3.4.6
  * `zlib` to 1.3.1
  * `sqlite` to 3.46.0.

Python modules updates:
  * `psutil` to 5.9.6 on generic glibc-based Linux,
  * `psutil` to 6.0.0 on the other platforms.

**Drive-by changes**:
 * `compat` tests are now disabled as the branch for Python 2.7 tests is unmaintained
 * macOS package is now built on macOS 13.


How to try and test the changes
===============================

reviewers: @adiroiban 

For a quick picture of the overall security situation per OS, check `external_deps.fods` in LibreOffice Calc.

To check other changes to our scripts and docs:
```
git diff master .github/ chevah_build  python-modules/chevah-python-test/ src/*/README
```

For the `cryptography` patch:
```
git diff master python-modules/cryptography*
```

For Python 2.7.18 patches:
```
git diff master src/python
```

For OpenSSL 1.1.1w patches:
```
git diff master src/openssl
```
